src/PeriodEnd/Refunds/SFA.DAS.ProviderPayments.Calc.Refunds/Services/LearnerRefundProcessor.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using NLog;
+using SFA.DAS.Payments.DCFS.Domain;
+using SFA.DAS.ProviderPayments.Calc.Refunds.Domain;
+using SFA.DAS.ProviderPayments.Calc.Refunds.Services.Dependiencies;
+using SFA.DAS.ProviderPayments.Calc.Shared.Infrastructure.Data.Entities;
+
+namespace SFA.DAS.ProviderPayments.Calc.Refunds.Services
+{
+    public class LearnerRefundProcessor : IProcessLearnerRefunds
+    {
+        private readonly ILogger _logger;
+
+        private static readonly List<FundingSource> FundingSourcesToRefund = new List<FundingSource>
+        {
+            FundingSource.Levy,
+            FundingSource.CoInvestedEmployer,
+            FundingSource.CoInvestedSfa,
+            FundingSource.FullyFundedSfa
+        };
+
+        public LearnerRefundProcessor(ILogger logger)
+        {
+            _logger = logger;
+        }
+
+        public List<PaymentEntity> ProcessRefundsForLearner(
+            List<RequiredPaymentEntity> refunds,
+            List<HistoricalPaymentEntity> previousPayments)
+        {
+            var refundGroups = refunds.ToLookup(x => new RefundGroup(x));
+            var volatilePreviousPayments = new List<HistoricalPaymentEntity>(previousPayments);
```

Comment:
Can we call this copyOfPreviousPayments as it's not volatile in the c# sense. You are not changing values only adding to a list